src/main/java/reposense/util/FileUtil.java

```diff
@@ -38,6 +41,71 @@
+     * @param configs contains the list of configuration information for a single repository
+     * @return a set of relevant repo folders (to be zipped later)
+     */
+    public static HashSet<Path> getReportFolders(Path sourcePath, List<RepoConfiguration> configs) {
+        HashSet<Path> relevantFolders = new HashSet<>();
+        for (RepoConfiguration repoConfiguration : configs) {
+            relevantFolders.add(Paths.get(sourcePath + File.separator + repoConfiguration.getDisplayName())
+                    .toAbsolutePath());
+        }
+        return relevantFolders;
+    }
+
+    /**
+     * Zips all the relative folders and relevant files of {@code fileTypes} contained in
+     * {@code sourceAndOutputPath} directory into the same folder.
+     * @param pathsToZip contains the folders and files to be zipped.
+     * @param sourceAndOutputPath contains the directory where the source folders and files are located at and
+     *                           where to be zipped to.
+     * @param fileTypes contains the file types to be zipped. Only files which are of the type "fileTypes" will be
+     *                  zipped.
+     */
+    public static void zipFoldersAndFiles(HashSet<Path> pathsToZip,
+            Path sourceAndOutputPath, String... fileTypes) {
+        zipFoldersAndFiles(pathsToZip, sourceAndOutputPath, sourceAndOutputPath, fileTypes);
+    }
+
+    /**
+     * Zips all the relevant files and relative folders
+     * @param pathsToZip contains the relevant folders to be zipped.
+     * @param sourcePath contains the directory where the relevant folders and files are located at
+     * @param outputPath contains the directory to be zipped to.
+     * @param fileTypes contains the file types to be zipped. Only files which are of the type "fileTypes" will be
+     *                  zipped.
+     */
+    public static void zipFoldersAndFiles(HashSet<Path> pathsToZip,
+            Path sourcePath, Path outputPath, String... fileTypes) {
+        try (
+                FileOutputStream fos = new FileOutputStream(outputPath + File.separator + ZIP_FILE);
+                ZipOutputStream zos = new ZipOutputStream(fos)
+        ) {
+            Set<Path> allFiles = getFilePaths(sourcePath, fileTypes);
+            for (Path path : allFiles) {
+                String filePath = sourcePath.relativize(path.toAbsolutePath()).toString();
+                if (pathsToZip.contains(path.getParent()) || pathsToZip.contains(path)) {
+                    String zipEntry = Files.isDirectory(path) ? filePath + File.separator : filePath;
+                    zos.putNextEntry(new ZipEntry(zipEntry.replace("\\", "/")));
+                    if (Files.isRegularFile(path)) {
+                        try (InputStream is = Files.newInputStream(path)) {
+                            int length;
+                            while ((length = is.read(buffer.array())) > 0) {
+                                zos.write(buffer.array(), 0, length);
```

Comment:
though its a legacy but since you're revamping it, do help to make the code more readable here, it is a rather huge [arrow head](https://blog.codinghorror.com/flattening-arrow-code/)